iOS/Flows/Conversation/Conversation Header/MembersViewController.swift

```diff
@@ -33,53 +32,64 @@ class MembersViewController: DiffableCollectionViewController<MembersCollectionV
         super.initializeViews()
 
         self.view.clipsToBounds = false
-        self.collectionView.clipsToBounds = false 
+        self.collectionView.clipsToBounds = false
 
-        self.collectionView.animationView.isHidden = true 
+        self.collectionView.animationView.isHidden = true
 
         ConversationsManager.shared.$activeConversation
             .removeDuplicates()
-            .mainSink { conversation in
-            Task {
-                guard let cid = conversation?.cid else {
-                    await self.dataSource.deleteAllItems()
-                    return
-                }
-                
-                self.conversationController = ChatClient.shared.channelController(for: cid)
-
-                await self.loadData()
-                self.subscribeToUpdates(for: conversation)
-                
-            }.add(to: self.taskPool)
-        }.store(in: &self.cancellables)
+            .mainSink { [unowned self] conversation in
+                Task {
+                    guard let cid = conversation?.cid else {
+                        await self.dataSource.deleteAllItems()
+                        return
+                    }
+
+                    let conversationController = ChatClient.shared.channelController(for: cid)
+                    self.conversationController = conversationController
+
+                    await self.loadData()
+
+                    guard !Task.isCancelled else { return }
+
+                    self.subscribeToUpdates(for: conversationController)
+                }.add(to: self.autocancelTaskPool)
+            }.store(in: &self.cancellables)
     }
 
-    func subscribeToUpdates(for conversation: Conversation?) {
-        self.conversationController?
+    /// The subscriptions for the current conversation.
+    private var conversationCancellables = Set<AnyCancellable>()
+
+    func subscribeToUpdates(for conversationController: ConversationController) {
+        // Clear out previous subscriptions.
+        self.conversationCancellables.removeAll()
```

Comment:
We were never cleaning up the conversation subscriptions so the members VC ended up getting a whole bunch of redundant events.